vue/components/ui/organisms/table-menu/table-menu.vue

```diff
@@ -0,0 +1,336 @@
+    width: 100%;
+}
+
+.table-menu .content-menu ::v-deep .content .table-title,
+.table-menu .content-menu ::v-deep .menu .menu-title {
+    color: $dark;
+    font-size: 18px;
+    font-weight: 500;
+    padding: 10px 0px 10px 0px;
+}
+
+.table-menu .content-menu ::v-deep .content .table-title {
+    padding: 10px 0px 10px 20px;
+}
+
+.table-menu .content-menu ::v-deep .content .table-content {
+    overflow-y: auto;
+}
+
+.table-menu .content-menu ::v-deep .content .table .table-head .table-column {
+    overflow: hidden;
+    text-overflow: ellipsis;
+}
+
+.table-menu .content-menu .content .table .table-body > tr {
+    cursor: pointer;
+    transition: opacity 0.1s ease-in-out;
+}
+
+.table-menu .content-menu .content .table .table-body > tr:hover {
+    background-color: $selected-color;
+}
+
+.table-menu .content-menu .content .table .table-body > tr.selected {
+    background-color: #e3e8f1;
+}
+
+.table-menu .content-menu .menu .button-color.delete-item {
+    margin: 20px 0px 20px 0px;
+}
+</style>
+
+<script>
+export const TableMenu = {
+    name: "table-menu",
+    props: {
+        items: {
+            type: Array,
+            default: () => []
+        },
+        columns: {
+            type: Array,
+            default: () => []
+        },
+        editColumns: {
+            type: Array,
+            default: () => []
+        },
+        selectedIndex: {
+            type: Number,
+            default: null
+        },
+        sort: {
+            type: String,
+            default: null
+        },
+        sortMethod: {
+            type: Function,
+            default: (items, column, reverse) => {
+                return items.sort((first, second) => {
+                    const order = reverse ? -1 : 1;
+                    const sort = first[column] > second[column];
+                    return order * (sort ? 1 : -1);
+                });
+            }
+        },
+        reverse: {
+            type: Boolean,
+            default: false
+        },
+        menuMode: {
+            type: String,
+            default: "collapse"
+        },
+        menuAlignment: {
+            type: String,
+            default: "right"
+        },
+        maxHeight: {
+            type: Number,
+            default: null
+        },
+        menuWidth: {
+            type: Number,
+            default: 300
+        },
+        menuTitle: {
+            type: String,
+            default: "Arguments"
+        },
+        tableTitle: {
+            type: String,
+            default: null
+        },
+        menuVisible: {
+            type: Boolean,
+            default: true
+        },
+        menuBackgroundColor: {
+            type: String,
+            default: "#ffffff"
+        },
+        inputVariant: {
+            type: String,
+            default: "dark"
+        },
+        animationDuration: {
+            type: Number,
+            default: 0.3

```

Comment:
> If this is the default value of the component. Set this default to null, and change the logic in the component to use its default when null is received.
> 
> We do this so that we dont need to change the default values in molecules and organisms, when changing an atom's default value.